pmd-plsql/etc/grammar/PLSQL.jjt

```diff
@@ -5133,6 +5137,7 @@ TOKEN [IGNORE_CASE]:
 <ROWTYPE: "ROWTYPE"> |
 <SAVE: "SAVE"> |
 <SAVEPOINT: "SAVEPOINT"> |
+<SCALAR: "SCALAR"> |
```

Comment:
SCALAR and SQL_MACRO are neither reserved words or keywords. I'll try to avoid defining extra tokens for something, that is not reserved/keywords. IIRC javacc has some limit of the number of token definitions, which we hit in the past. Unfortunately, everything was defined as token in the past in this grammar, which created lots of problems - especially as these tokens can be used as identifiers in other contexts.

I'll therefore remove these from here and use directly `"SCALAR"` where it is used and also make sure to add these two tokens to KEYWORD_NOT_RESERVED, so that they still can be used as identifiers.